### PR TITLE
cgosqlite: cleanup the import layout some more

### PR DESCRIPTION
**Pull request by raggi**
Just removes duplicate imports and orders them better.

### CODE REVIEW

File: cgosqlite/cgosqlite.h [hunk 1]
```diff
@@ -1,4 +1,6 @@
-#include <stdint.h>
```
Comment (creachadair) [inline cgosqlite/cgosqlite.h:1]: I think we'll still need `<stdint.h>` here.

File: cgosqlite/cgosqlite.go [hunk 2]
```diff
@@ -33,21 +33,7 @@ package cgosqlite
 // // Select POSIX 2014 at least for clock_gettime.
 // #cgo CFLAGS: -D_XOPEN_SOURCE=600
 // #cgo CFLAGS: -D_DARWIN_C_SOURCE=1
-// #cgo linux CFLAGS: -std=c99
 //
-// // On Android, unlike Linux, there are no separate libpthread or librt
```
Comment (creachadair) [inline cgosqlite/cgosqlite.go:38]: Some of the imports are redundant, but are we sure that the other compiler directives are too? I'm not sure the tests in this repo are exercising that.

Comment (raggi) [reply cgosqlite/cgosqlite.go:38]: we pass `-std=gnu11` in gocross, the `-std=c99` is somewhat redundant. sqlite itself still builds with c89 and only requires a few memory functions and an allocator, but turns out that fts5 is the thing that depends on libm.

Comment (raggi) [reply cgosqlite/cgosqlite.go:38]: (but on clang systems, log(), which it needs is provided without needing to link anything)

**Review by creachadair — APPROVED** [review]

**Pull request merged**